### PR TITLE
Signature cleanup

### PR DESCRIPTION
**Pull request by debris**
based on #1851 

- this pr unifies duplicated (in util and ethstore) crypto functions and structs

### CODE REVIEW

File: ethcrypto/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+/// ECDH functions
+#[cfg_attr(feature="dev", allow(similar_names))]
+pub mod ecdh {
+	use secp256k1::{ecdh, key};
+	use ethkey::{Secret, Public, SECP256K1};
+	use Error;
+
+	/// Agree on a shared secret
+	pub fn agree(secret: &Secret, public: &Public) -> Result<Secret, Error> {
+
```
Comment (NikVolf) [inline ethcrypto/src/lib.rs]: redundant line

**Comment by gavofyork** [comment]
fails travis.

**Comment by debris** [comment]
travis fixed

**Comment by gavofyork** [comment]
can't merge.

**Pull request merged**